pkg/ipam/allocator/multipool/pool_allocator.go

```diff
@@ -770,3 +853,62 @@ func (p *PoolAllocator) releaseOrphanCIDR(targetNode, sourcePool string, cidr ne
 
 	return nil
 }
+
+func (c *cidrPool) countAllocatedIPsInPrefix(cidr netip.Prefix, isV6 bool) (*big.Int, error) {
+	getIndices := func(cidr netip.Prefix, maskSize int) (begin, end int, err error) {
+		if !cidr.IsValid() {
+			return -1, -1, fmt.Errorf("invalid CIDR: %v", cidr)
+		}
+
+		prefixBits := cidr.Bits()
+		if prefixBits < maskSize {
+			return -1, -1, fmt.Errorf("CIDR mask size %d is smaller than the pool mask size %d", prefixBits, maskSize)
+		}
+
+		begin = 0
+		end = (1 << (prefixBits - maskSize)) - 1
```

Comment:
I don't get the logic here. Mask size must be less than the prefix bits, since in multi pool IPAM we add CIDRs (e.g: 1.1.1.0/24 -> prefixBits == 24) that the operator will divide in smaller CIDRs of the same masksize (e.g: 1.1.1.0/28, 1.1.1.16/28 and so on -> maskSize == 28) and assign them to each agent. Therefore `prefixBits < maskSize` will always be true.